src/rgw/driver/rados/rgw_bucket.cc

```diff
@@ -2132,6 +2132,83 @@ int RGWMetadataHandlerPut_Bucket::put_post(const DoutPrefixProvider *dpp)
   return ret;
 }
 
+int update_bucket_topic_mappings(const DoutPrefixProvider* dpp,
+                                 RGWBucketCompleteInfo* orig_bci,
+                                 RGWBucketCompleteInfo* current_bci,
+                                 rgw::sal::Driver* driver) {
+  const auto decode_attrs =
+      [&](const rgw::sal::Attrs& attrs,
```

Comment:
why do you need to capture everything by ref?
looks like you are using only the arguments to the lambda?

Reply:
> why do you need to capture everything by ref? looks like you are using only the arguments to the lambda?

Done.